app/controllers/users_controller.rb

```diff
@@ -1,18 +1,55 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!
   before_action :set_user
 
   def set_password; end
 
   def update_set_password
     if @user.update(user_password_params)
+      @user.user!
       redirect_to home_index_path
     else
       render :set_password
     end
   end
 
+  def show; end
+
+  def edit; end
+
+  def update 
```

Comment:
Please check if we can update it in following way

def update
  @user.skip_validations = user_params[:password].blank? && user_params[:password_confirmation].blank?
  @user.skip_confirmation! if @user.skip_validations?
  updated =
      if @user.skip_validations
        @user.update_without_password(user_params.except(:current_password))
      else
        @user.update(user_params)
      end

  return render 'show' if updated

  render 'edit'
end